python/src/scipp/compat/xarray_compat.py

```diff
@@ -0,0 +1,97 @@
+import typing
+
+import xarray
+import scipp
+
+
+def from_xarray(xarray_obj: typing.Union[xarray.DataArray, xarray.Dataset]):
+    """
+    Convenience method to convert an xarray object into the corresponding
+    scipp object.
+
+    If you know in advance what type of object you need to convert, you can
+    also call from_xarray_dataarray or from_xarray_dataset directly.
+
+    :param xarray_obj: the xarray object to convert; must be either an xarray
+        DataArray or Dataset object.
+    :return: the converted scipp object.
+    """
+    if isinstance(xarray_obj, xarray.DataArray):
+        return from_xarray_dataarray(xarray_obj)
+    elif isinstance(xarray_obj, xarray.Dataset):
+        return from_xarray_dataset(xarray_obj)
+    else:
+        raise ValueError(
+            f"from_xarray: cannot convert type '{type(xarray_obj)}'")
+
+
+def from_xarray_dataarray(
+        xarray_dataarray: xarray.DataArray) -> scipp.DataArray:
```

Comment:
Minor suggestion here and below:
```suggestion
        da: xr.DataArray) -> scipp.DataArray:
```
is the commonly adopted naming (`import xarray as xr`), `da` for data arrays, `ds` for datasets.